test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -2430,4 +2433,40 @@ public static <T extends Throwable> T expectThrows(Class<T> expectedType, Reques
             () -> builder.get().decRef() // dec ref if we unexpectedly fail to not leak transport response
         );
     }
+
+    /**
+     * Run {@code numberOfTasks} parallel tasks that were created by the given {@code taskFactory}. On of the tasks will be run on the
+     * calling thread, the rest will be run on a new thread.
+     * @param numberOfTasks number of tasks to run in parallel
+     * @param taskFactory task factory
+     */
+    public static void runInParallel(int numberOfTasks, IntConsumer taskFactory) throws InterruptedException, ExecutionException {
```

Comment:
We don't throw `ExecutionException` any more (which has some consequences on callers)

```suggestion
    public static void runInParallel(int numberOfTasks, IntConsumer taskFactory) throws InterruptedException {
```